web/client/plugins/queryPanelWithMap/SmartQueryForm.jsx

```diff
@@ -0,0 +1,148 @@
+        spatialPanelExpanded: state.queryform.spatialPanelExpanded,
+        featureTypeConfigUrl: state.query && state.query.url,
+        searchUrl: state.query && state.query.url,
+        featureTypeName: state.query && state.query.typeName,
+        ogcVersion: "1.1.0",
+        params: {typeName: state.query && state.query.typeName},
+        resultTitle: "Query Result",
+        showGeneratedFilter: false,
+        allowEmptyFilter: true,
+        emptyFilterWarning: true,
+        maxHeight: state.map && state.map.present && state.map.present.size && state.map.present.size.height,
+        zoom: (mapSelector(state) || {}).zoom,
+        projection: (mapSelector(state) || {}).projection
+    };
+}, dispatch => {
+    return {
+        attributeFilterActions: bindActionCreators({
+            onAddGroupField: addGroupField,
+            onAddFilterField: addFilterField,
+            onRemoveFilterField: removeFilterField,
+            onUpdateFilterField: updateFilterField,
+            onUpdateExceptionField: updateExceptionField,
+            onUpdateLogicCombo: updateLogicCombo,
+            onRemoveGroupField: removeGroupField,
+            onChangeCascadingValue: changeCascadingValue,
+            toggleMenu: toggleMenu,
+            onExpandAttributeFilterPanel: expandAttributeFilterPanel
+        }, dispatch),
+        spatialFilterActions: bindActionCreators({
+            onChangeSpatialFilterValue: changeSpatialFilterValue,
+            onExpandSpatialFilterPanel: expandSpatialFilterPanel,
+            onSelectSpatialMethod: selectSpatialMethod,
+            onSelectViewportSpatialMethod: selectViewportSpatialMethod,
+            onSelectSpatialOperation: selectSpatialOperation,
+            onChangeDrawingStatus: changeDrawingStatus,
+            onRemoveSpatialSelection: removeSpatialSelection,
+            onShowSpatialSelectionDetails: showSpatialSelectionDetails,
+            onChangeDwithinValue: changeDwithinValue,
+            zoneFilter: zoneGetValues,
+            zoneSearch,
+            zoneChange
+        }, dispatch),
+        queryToolbarActions: bindActionCreators({
+            onQuery: search,
+            onReset,
+            onSaveFilter: storeCurrentFilter,
+            onRestoreFilter: discardCurrentFilter,
+            storeAppliedFilter: applyFilter,
+            onChangeDrawingStatus: changeDrawingStatus
+
+        }, dispatch),
+        crossLayerFilterActions: bindActionCreators({
+            expandCrossLayerFilterPanel,
+            setCrossLayerFilterParameter,
+            addCrossLayerFilterField,
+            updateCrossLayerFilterField,
+            removeCrossLayerFilterField,
+            resetCrossLayerFilter,
+            toggleMenu: (rowId, status) => toggleMenu(rowId, status,  "crossLayer")
+        }, dispatch),
+        controlActions: bindActionCreators({onToggleQuery: toggleControl.bind(null, 'queryPanelWithMap', null)}, dispatch)
```

Comment:
Also in this case the difference is only this controlAction (that may be unified) and maybe the diffferent imports.

Reply:
I mean we can use the `queryPanel` control in any case, when we have one plugin only